### PR TITLE
(Issue #116) Added table_prefix to model generator form.

### PR DESCRIPTION
**Pull request by umanamente**
Now the **Model Class** field on the client side is generated correctly. (Issue #116)

### CODE REVIEW

File: assets/gii.js [hunk 1]
```diff
@@ -201,6 +201,17 @@ yii.gii = (function ($) {
             // model generator: translate table name to model class
             $('#model-generator #generator-tablename').on('blur', function () {
                 var tableName = $(this).val();
+                var tablePrefix = $(this).attr('table_prefix');
+                if (typeof tablePrefix === typeof undefined || tablePrefix === false) {
+                    tablePrefix = '';
```
Comment (SilverFire) [inline assets/gii.js]: Might be simplified to `var tablePrefix = $(this).attr('table_prefix') || '';`

File: assets/gii.js [hunk 2]
```diff
@@ -201,6 +201,17 @@ yii.gii = (function ($) {
...
+                }
+                if(tablePrefix.length > 0){
```
Comment (SilverFire) [inline assets/gii.js]: `if (`

File: assets/gii.js [hunk 3]
```diff
@@ -201,6 +201,17 @@ yii.gii = (function ($) {
...
+                    // if starts with prefix
+                    if(tableName.slice(0, tablePrefix.length) == tablePrefix){
```
Comment (SilverFire) [inline assets/gii.js]: `==` -> `===`

File: generators/model/Generator.php [hunk 4]
```diff
@@ -181,6 +181,19 @@ public function stickyAttributes()
     }
 
     /**
+     * Returns db tables prefix
+     *
+     */
+    public function getTablePrefix(){
```
Comment (SilverFire) [inline generators/model/Generator.php]: `{` should be on the newline

File: generators/model/form.php [hunk 5]
```diff
@@ -6,7 +6,7 @@
 /* @var $form yii\widgets\ActiveForm */
 /* @var $generator yii\gii\generators\form\Generator */
 
-echo $form->field($generator, 'tableName');
+echo $form->field($generator, 'tableName')->textInput(array('table_prefix' => $generator->getTablePrefix()));
```
Comment (SilverFire) [inline generators/model/form.php]: We're using short array syntax `[1, 2, 3]`

File: generators/model/Generator.php [hunk 6]
```diff
@@ -181,6 +181,19 @@ public function stickyAttributes()
     }
 
     /**
+     * Returns db tables prefix
+     *
```
Comment (SilverFire) [inline generators/model/Generator.php]: We need a `@return` PHPDoc directive

**Comment by SilverFire** [comment]
Looks good

File: assets/gii.js [hunk 7]
```diff
@@ -201,6 +201,14 @@ yii.gii = (function ($) {
             // model generator: translate table name to model class
             $('#model-generator #generator-tablename').on('blur', function () {
                 var tableName = $(this).val();
+                var tablePrefix = $(this).attr('table_prefix') || '';
+                if (tablePrefix.length > 0) {
```
Comment (samdark) [inline assets/gii.js]: Сould be just `if (tablePrefix.length)`

**Comment by SilverFire** [comment]
Add a CHANGELOG line and it's ready for merge. Thank you

**Comment by SilverFire** [comment]
As a note.
We have a checkbox "Use Table Prefix". I've firstly thought, that the changes in this PR should affect the `Model Name` input only when the checkbox is ON. But then I realized, that my idea is wrong, because the class name should be named correctly anyway

**Comment by SilverFire** [comment]
Merged with minor adjustments https://github.com/yiisoft/yii2-gii/commit/402902fee3e5b4073aa98b2baa0414c6f38e0818

**Comment by SilverFire** [comment]
Thank you!

**Pull request closed without merging**